ViAn/Video/frameprocessor.cpp

```diff
@@ -170,20 +170,23 @@ void FrameProcessor::process_frame() {
         cv::rotate(manipulated_frame, manipulated_frame, m_rotate_direction);
     }
 
-    // Create zoom perview mat
+    // Create zoom preview mat
     //    cv::Mat tmp = manipulated_frame.clone();
     //    double factor{0.5};
     //    cv::resize(tmp, tmp, cv::Size(), factor, factor);
     //    cv::Rect view_rectangle = m_zoomer.get_view_rect();
     //    cv::rectangle(tmp, view_rectangle.tl() * factor, view_rectangle.br() * factor, cv::Scalar(255,255,255), 2);
     //    cv::imshow("test", tmp); // can cause a deadlock
 
-    // Draws the overlay
+    // Draws the text drawings on the overlay
     m_overlay->draw_overlay(manipulated_frame, m_frame_index->load());
 
     // Apply zoom
     m_zoomer.scale_frame(manipulated_frame);
 
+    // Draws the other drawings on the overlay
+    m_overlay->draw_overlay_scaled(manipulated_frame, m_frame_index->load(), Utility::from_qpoint(m_z_settings->anchor), m_z_settings->zoom_factor);
```

Comment:
There is a slight risk that `m_frame_index->load()` could cause an issue here. It's an atomic action and we use it quite freely without locks. Therefore the user could potentially change the frame number after the frame has been loaded and thus the program would draw the wrong drawings. Would probably be good to take a look at.